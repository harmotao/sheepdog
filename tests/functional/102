#!/bin/bash

# Test object creation for store_driver(tree)

. ./common

for i in 0 1 2 3 4 5; do
    _start_sheep $i
done
_wait_for_sheep 6
_cluster_format -c 3 -b tree

echo "CHECK data/ledger object"

echo "create inode/data/ledger objects"
$DOG vdi create test 16M
_random | $DOG vdi write test 0 512
$DOG vdi snapshot test
_random | $DOG vdi write test 0 512
$DOG vdi clone -s 1 test test2
_random | $DOG vdi write test2 0 512
$DOG vdi delete test
$DOG vdi list | _filter_short_date

_node_info

echo "after delete whole vdis"
$DOG vdi delete test -s 1
$DOG vdi delete test2

_node_info

echo "yes" | $DOG cluster format -b tree

echo "CHECK erasure object"
echo "create inode/data erasure code objects"

$DOG vdi create -c 4:2 test 16M
_random | $DOG vdi write test 0 512
$DOG vdi list | _filter_short_date

_node_info

$DOG vdi delete test

echo "yes" | $DOG cluster format -b tree

echo "CHECK attr object"
echo -n value > $STORE/tmp.dat

$DOG vdi create test 16M
$DOG vdi setattr test key value
$DOG vdi getattr test key | diff - $STORE/tmp.dat

echo "create attr objects"
_node_info

$DOG vdi delete test
