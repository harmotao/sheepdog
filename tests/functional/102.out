QA output created by 102
using backend tree store
CHECK data/ledger object
create inode/data/ledger objects
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1   16 MB  4.0 MB  0.0 MB DATE   7c2b25      3                22
c test2        0   16 MB  4.0 MB  0.0 MB DATE   fd3815      3                22
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	1	2	0	0	1	0
1	1	3	0	0	0	0
2	1	3	0	0	0	0
3	2	0	0	0	0	0
4	1	1	0	0	1	0
5	0	0	0	0	1	0
after delete whole vdis
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	0	2	0	0	0	0
1	0	3	0	0	0	0
2	0	3	0	0	0	0
3	0	0	0	0	0	0
4	0	1	0	0	0	0
5	0	0	0	0	0	0
    __
   ()'`;
   /\|`
  /  |   Caution! The cluster is not empty.
(/_)_|_  Are you sure you want to continue? [yes/no]: using backend tree store
CHECK erasure object
create inode/data erasure code objects
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0   16 MB  4.0 MB  0.0 MB DATE   7c2b25    4:2                22
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	1	1	0	0	0	0
1	1	1	0	0	0	0
2	1	1	0	0	0	0
3	1	1	0	0	0	0
4	1	1	0	0	0	0
5	1	1	0	0	0	0
    __
   ()'`;
   /\|`
  /  |   Caution! The cluster is not empty.
(/_)_|_  Are you sure you want to continue? [yes/no]: using backend tree store
CHECK attr object
create attr objects
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	0	0	0	1	0	0
1	0	1	0	1	0	0
2	0	1	0	0	0	0
3	0	0	0	0	0	0
4	0	1	0	0	0	0
5	0	0	0	1	0	0
